job_runner/reporting/interactive.py

```diff
@@ -1,20 +1,21 @@
-from job_runner.reporting.message import Finish, Start
+from job_runner.reporting.message import Message, Finish, Start
 from job_runner.reporting.base import Reporter
 
 
 class Interactive(Reporter):
-    def report(self, msg):
+    def _report(self, msg: Message) -> str:  # type: ignore
```

Comment:
What's the ignore for?

Reply:
There is no reason, it used to be when I was missing type hints. but I forgot to remove it

Reply:
ow wait, mistake, this is for pylint not mypy `job_runner/reporting/interactive.py:6:4: R1710: Either all return statements in a function should return an expression, or none of them should. (inconsistent-return-statements)`

Reply:
it's because of the explicit 
```python
if ...:
  return
if ...:
  return
```
instead of a `if... else' block or `if ..: return ... return`

Reply:
So ignore will stay

Reply:
I removed it, but it fails on the none-strict version of mypy configuration

Reply:
What is the failure? I can't reproduce that.

Running `mypy -p ert -p ert3`

Reply:
That's interested unable to reproduce it locally, but it does fail on the pipeline.
Anyhow `job_runner/reporting/interactive.py:6: error: Missing return statement` is the error I will push it without it to show the problem

Reply:
You have outdated packages. Do `pip install --upgrade types-six typing-extensions` or something like it and you'll see the error.

You're missing a return statement for `_report`, because your checks on `isinstance(msg, type)` isn't exhaustive, so mypy expects to find some return at the end of the function.

Probably be easier to revert, or else you have to change the return signature to be `Optional[str]`, return `None` and then update `report` to either print or not…